### PR TITLE
Create a singleton web-socket client - Closes #4411, #4442

### LINKED ISSUE #4411: Create websocket singleton client
### Description

Create a single WebSocket connection by the client

### Acceptance Criteria

- Allow the application to use one single instance of connection
- Should handle connection, re-connection
- Close the connection before re-connection
- Integrate with lisk sdk is required
- Add required unit tests. 

### Good to have
Use Lisk SDK v6 lisk APIClient 

`const { createWSClient } from '@liskhq/lisk-api-client';`

### PR DESCRIPTION
**Pull request by soroushm**
### What was the problem?

- web-socket has multiple network connection
- RPC support for react query 
- fetch method dosn't work on test env

### How was it solved?
by create a singleton socket client,
by adding RPC function
by create a singleton axios instance

### How was it tested?
by unit test

### CODE REVIEW

File: src/modules/blockchainApplication/manage/hooks/useCurrentApplication.js [hunk 1]
```diff
@@ -13,8 +14,9 @@ export function useCurrentApplication() {
   const setApplication = useCallback(
     (application) => {
       dispatch(setCurrentApplication(application));
-      // Set default node
-      setCurrentNode(application.apis[0]);
+      /* istanbul ignore next */
```
Comment (clemente-xyz) [inline src/modules/blockchainApplication/manage/hooks/useCurrentApplication.js:17]: `istanbul` ?

File: src/utils/api/rpc.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+const rpc = ({
+  event,
+  params,
+}) => new Promise((resolve, reject) => {
+  if (socket.client.disconnected) {
+    reject(new Error('socket not connected'));
+    return;
+  }
+  socket.client.emit(event, params || {}, (response) => {
+    if (Object.keys(response).length && response.error) {
```
Comment (clemente-xyz) [inline src/utils/api/rpc.js:23]: you can just do `response?.error`

File: src/utils/api/socket.test.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  //     socket.create({ baseUrl: `http://localhost:${port}` });
+  //     // socket.client.on('connect', done);
+  //     socket.client.on('connect', done);
+  //   });
+  // });
+  //
+  // afterAll(async () => {
+  //   await server.close();
+  //   await io.close();
+  // });
```
Comment (clemente-xyz) [inline src/utils/api/socket.test.js]: can you remove this commented code if its not being used?

**Review by clemente-xyz — CHANGES_REQUESTED** [review]
Great! just a few comments

File: package.json [hunk 4]
```diff
@@ -188,14 +188,14 @@
...
   "devDependencies": {
-    "@chainsafe/blst": "0.2.4",
     "@babel/core": "7.13.16",
     "@babel/plugin-proposal-nullish-coalescing-operator": "7.13.8",
     "@babel/plugin-proposal-optional-chaining": "7.13.12",
     "@babel/plugin-syntax-import-meta": "7.10.4",
     "@babel/plugin-transform-runtime": "7.13.15",
     "@babel/preset-env": "7.13.15",
     "@babel/preset-react": "7.13.13",
+    "@chainsafe/blst": "0.2.4",
```
Comment (clemente-xyz) [inline package.json]: i dont see this package imported anywhere. Why did you installed it?

**Review by clemente-xyz — CHANGES_REQUESTED** [review]

File: src/modules/blockchainApplication/manage/hooks/useCurrentApplication.js (hunk 1 above)
Comment (soroushm) [reply src/modules/blockchainApplication/manage/hooks/useCurrentApplication.js:17]: its for the test coverage

File: src/utils/api/socket.test.js (hunk 3 above)
Comment (soroushm) [reply src/utils/api/socket.test.js]: no i want to keep it we had issues to running this it need to be fix later

File: package.json (hunk 4 above)
Comment (soroushm) [reply package.json]: its just change the order it's used by sdk v6

**Review by eniolam1000752 — APPROVED** [review]

**Review by ikem-legend — APPROVED** [review]

**Review by clemente-xyz — APPROVED** [review]

**Pull request merged**